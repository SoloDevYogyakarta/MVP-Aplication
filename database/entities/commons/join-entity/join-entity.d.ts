import { Model, Optional } from 'sequelize';
import { BaseEntity } from '../../base-entity/base-entity';
export declare class JoinEntity extends BaseEntity {
    source_id: string;
    foreign_id: string;
}
export type JoinCreationAttribute = Optional<JoinEntity, 'id'>;
export interface JoinInstance extends Model<JoinCreationAttribute, JoinEntity>, JoinEntity {
}
export declare const joinEntity: import("sequelize").ModelCtor<JoinInstance>;
