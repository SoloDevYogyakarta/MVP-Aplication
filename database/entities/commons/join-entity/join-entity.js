"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.joinEntity = exports.JoinEntity = void 0;
const nanoid_1 = require("nanoid");
const sequelize_1 = require("sequelize");
const base_entity_1 = require("../../base-entity/base-entity");
const entity_1 = require("../../entity");
class JoinEntity extends base_entity_1.BaseEntity {
}
exports.JoinEntity = JoinEntity;
exports.joinEntity = entity_1.sequelize.define('JOIN', {
    public_id: sequelize_1.DataTypes.STRING,
    source_id: sequelize_1.DataTypes.STRING,
    foreign_id: sequelize_1.DataTypes.STRING,
}, {
    tableName: 'JOIN',
    schema: 'COMMONS',
    timestamps: false,
    hooks: {
        beforeCreate(attributes, options) {
            const instance = attributes;
            instance.public_id = (0, nanoid_1.nanoid)();
        },
    },
});
//# sourceMappingURL=join-entity.js.map