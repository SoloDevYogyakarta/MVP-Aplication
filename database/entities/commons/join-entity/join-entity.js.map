{"version":3,"file":"join-entity.js","sourceRoot":"","sources":["../../../../../src/database/entities/commons/join-entity/join-entity.ts"],"names":[],"mappings":";;;AAAA,mCAAgC;AAChC,yCAAwE;AACxE,+DAA2D;AAC3D,yCAAyC;AAEzC,MAAa,UAAW,SAAQ,wBAAU;CAGzC;AAHD,gCAGC;AAMY,QAAA,UAAU,GAAG,kBAAS,CAAC,MAAM,CACxC,MAAM,EACN;IACE,SAAS,EAAE,qBAAS,CAAC,MAAM;IAC3B,SAAS,EAAE,qBAAS,CAAC,MAAM;IAC3B,UAAU,EAAE,qBAAS,CAAC,MAAM;CACI,EAClC;IACE,SAAS,EAAE,MAAM;IACjB,MAAM,EAAE,SAAS;IACjB,UAAU,EAAE,KAAK;IACjB,KAAK,EAAE;QACL,YAAY,CAAC,UAAU,EAAE,OAAO;YAC9B,MAAM,QAAQ,GAAG,UAAU,CAAC;YAC5B,QAAQ,CAAC,SAAS,GAAG,IAAA,eAAM,GAAE,CAAC;QAChC,CAAC;KACF;CACF,CACF,CAAC"}