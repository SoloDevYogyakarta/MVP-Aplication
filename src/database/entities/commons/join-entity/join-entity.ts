import { nanoid } from 'nanoid';
import { DataTypes, Model, ModelAttributes, Optional } from 'sequelize';
import { BaseEntity } from '../../base-entity/base-entity';
import { sequelize } from '../../entity';

export class JoinEntity extends BaseEntity {
  source_id: string;
  foreign_id: string;
}

export type JoinCreationAttribute = Optional<JoinEntity, 'id'>;
export interface JoinInstance
  extends Model<JoinCreationAttribute, JoinEntity>,
    JoinEntity {}
export const joinEntity = sequelize.define<JoinInstance>(
  'JOIN',
  {
    public_id: DataTypes.STRING,
    source_id: DataTypes.STRING,
    foreign_id: DataTypes.STRING,
  } as ModelAttributes<JoinInstance>,
  {
    tableName: 'JOIN',
    schema: 'COMMONS',
    timestamps: false,
    hooks: {
      beforeCreate(attributes, options) {
        const instance = attributes;
        instance.public_id = nanoid();
      },
    },
  },
);
