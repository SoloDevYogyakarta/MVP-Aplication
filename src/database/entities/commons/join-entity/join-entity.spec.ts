import { ModelCtor } from 'sequelize';
import { joinEntity, JoinInstance } from './join-entity';

describe('JoinEntity', () => {
  let entity: ModelCtor<JoinInstance>;

  beforeEach(() => {
    entity = joinEntity;
  });

  it('should to be defined', () => expect(entity).toBeDefined());

  it('render correctly', () => expect(entity).toMatchSnapshot());
});
