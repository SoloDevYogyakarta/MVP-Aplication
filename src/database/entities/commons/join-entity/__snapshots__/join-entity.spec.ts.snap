// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JoinEntity render correctly 1`] = `[Function]`;
